components/molecules/client/filter-group/component.tsx

```diff
@@ -0,0 +1,89 @@
+'use client';
+
+import React from 'react';
+import { Tag } from '@/components/atoms/tag';
+import { cn } from '@/lib/cn';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+
+
+interface FilterGroupProps {
+  options: string[];
+  onChange?: (selectedOptions: string[]) => void;
+  className?: string;
+}
+
+export const FilterGroup: React.FC<FilterGroupProps> = ({ options, onChange, className }) => {
+  const [selectedOptions, setSelectedOptions] = React.useState<string[]>([]);
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  // Initialize the selected options from query parameters
+  React.useEffect(() => {
+    const params = new URLSearchParams(searchParams);
+    const tagsFromUrl = params.getAll('tags');
```

Comment:
pass in the key of the url param as a prop so you can choose what the name of the filter is (e.g. ?topics=thing1&topics=thing2&authors=author1&authors=author2)